challenges-41-44.yml

```diff
@@ -0,0 +1,127 @@
+    > The reason lies within the "loosing reactivity" theme.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          Because there is a typo in `healthyError` and the template is rendering an undefined
+          value!
+        b: |
+          Because by adding or removing properties to objects in Vue 2, we are loosing
+          reactivity. This means that Vue cannot detect that a change has been made to
+          our object.
+        c: |
+          Because we're using a data object as opposed to a computed property. Using a computed
+          property would solve our issue.
+        d: |
+          Because we forgot to define the object properties as "reactive" when initializing
+          our component.
+      correct: b
+
+extending_loosing_reactivity2:
+  question: |
+    Given the following `validationErrors` object in my form component:
+
+    ```javascript
+    data() {
+        return {
+            validationErrors: {},
+        };
+    },
+    ```
+
+    And the fact that I need to validate on the following fields: `healthyError`, 'firstName',
+    'lastName' and 'email'.
+
+    I want to define my object from the get go, so as not to loose reactivity later, when
+    I set or unser error messages. The resutl is as follows:
+
+    ```javascript
+    data() {
+        return {
+            validationErrors: {
+                healthyError: null,
+                firstName: null,
+                lastName: null,
+                email: null,
+            },
+        };
+    },
+    ```
+
+    Why is this a needed first step towards fixing my reactivity issues?
+  explanation: |
+    In this case, setting all object properties from the get go is a very important step
+    so as to not have reactivity issues later. Just remember not to re-assign the entire
+    `validationErrors` object later!
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          While this won't solve *all* our issues, it its a very important first
+          step in avoiding the reactivity issues of adding/removing object propertoes
+          in Vue 2. In here, I'm setting all properties to `null` instead of not
+          setting them at all.
+        b: |
+          This fixes reactivity issues because I'm not reassigning this object anywhere
+          else in the code. It's just good practice to refine a "default" state for our
+          data objects in advance, so other programmers know what our code looks like.
+        c: |
+          Trick question! This does *not* fix any reactivity issues. To do so, we need to
+          configure all of these properties as "reactive".
+        d: |
+          This fixes reactivity issues only if we later use a computed property to
+          render the error messages. It is the first step of a combination of
+          techniques that uses a data object as the initial setup, plus a
+          computed property to deliver the actual error messages when needed.
+      correct: a
```

Comment:
I think this last question may not be needed - it feels VERY similar to the previous one. I think it's ok to not have it.

Reply:
Understood!